tests/test_assembly.py

```diff
@@ -808,20 +848,41 @@ def test_assemble_snp_bubble_stopbf(self, snp_bubble_structure):
 
         # do the labeling, but block the mutant with stop_bf
         stop_bf.count(mutant[HDN_L.pos + 1:HDN_L.pos + K + 1])
-        paths = lh.assemble_labeled_path(wildtype[:K], stop_bf)
-        assert len(paths) == 1
+        paths = asm.assemble(wildtype[:K], stop_bf)
 
+        assert len(paths) == 1
         assert any(utils._equals_rc(path, wildtype) for path in paths)
 
     # @pytest.mark.skip(reason='destroys your computer and then the world')
     def test_assemble_tandem_repeats(self, tandem_repeat_structure):
         # assemble one copy of a tandem repeat
         graph, repeat, tandem_repeats = tandem_repeat_structure
-
         lh = khmer._GraphLabels(graph)
+        asm = khmer.SimpleLabeledAssembler(lh)
+        paths = asm.assemble(repeat[:K])
 
-        paths = lh.assemble_labeled_path(repeat[:K])
         assert len(paths) == 1
         # There are K-1 k-mers spanning the junction between
         # the beginning and end of the repeat
         assert len(paths[0]) == len(repeat) + K - 1
+
+
+class TestJunctionCountAssembler:
+
+    def test_beginning_to_end_across_tip(self, right_tip_structure):
+        # assemble entire contig, ignoring branch point b/c of labels
```

Comment:
The general idea behind the labeled assembler, of which this is an optimized version, is to use known paths to connect across high degree nodes.  One primary use case is where we want to thread the graph with reads, so that we can resolve across repetitive k-mers.

A bit more background: De Bruijn graphs break input data up into k-mers, and this loses the longer range information present in any input data.  For example, if you used k=31 and input a 1kb contig, the fact that the 1kb contig was actually a known valid path through the data would be lost because you were breaking the contig up into 31-mers.  If the 31-mers form a path that doesn't branch, then we're fine - but of course in any real data set there will be high-degree nodes formed by repeats and sequencing errors.

There are a number of approaches to dealing with this, but the approach that @camillescott has implemented (first in the labelled assembler and then in the JunctionCount assembler) is to track k-mers across these high degree nodes. The labelled assembler does so by labeling the graph nodes on either side of the high degree node with the same label; the junction count assembler uses a trickier approach that is an optimized version of the same idea.

Please correct me if I'm wrong, @camillescott :)